### PR TITLE
Track handling

### PR DESCRIPTION
**Pull request by tkurki**
Store & retrieve track from InfluxDb:
- [x] save positions at 1 Hz
- [x] add route to retrieve track with relative time and specified resolution: /signalk/vX/self/track?timespan=1h&resolution=1m

Influxdb supported duration units:
image

### CODE REVIEW

**Comment by tkurki** [comment]
@sbender9 this could probably use another pair of eyes / somebody trying this out...

**Comment by sbender9** [comment]
Tested using a recored trip and have WilhelmSK showing the track on the map. 

Something strange that if I ask for 1s resolution I don't get any data back.

File: README.md [hunk 1]
```diff
@@ -11,6 +11,24 @@ The plugin writes only `self` data. It converts Signal K paths to InfluxDb `meas
...
+
+The plugin creates `/signalk/vX/api/self/track` endpoint that accepts two parameters:
+- timespan in the xxxY format, where xxx is a Number and Y one of 
+  - s  (seconds)
+  - m  (minutes)
+  - h  (hours)
+  - d  (days)
+  - w  (weeks)
+- resolution in the same format
+and returns GeoJSON MultiLineString. For example `http://localhost:3000/signalk/v1/api/self/track?timespan=1d&resolution=1` will return the data for the last 1 day (24 hours) with one position per hour. The data is simply sampled with InfluxDB's `first()` function.
```
Comment (sbender9) [inline README.md]: Example does not have the units `resolution=1`

**Comment by sbender9** [comment]
It would be nice to have a simple switch to turn track recording on and off.

**Comment by tkurki** [comment]
Yep, the plugin config ui is not what you would call simple. 

Other plugins would also need better custom ui's, like sails configuration.

Sounds like an extension point to provide plugin specific config panels is in order.

**Comment by tkurki** [comment]
http://localhost:3000/signalk/v1/api/self/track?timespan=1h&resolution=1s is working ok for me.

**Comment by sbender9** [comment]
It's http://localhost:3000/signalk/v1/api/self/track?timespan=1d&resolution=1s that's not working for me.

**Comment by tkurki** [comment]
Are you seeing any errors? 

Are you sure it is not your browser taking very long to format the result or something similar? Try with curl.

Works on my machine and my data, but I only have data here for some periods => the result set is not very large, which it would be with 24 * 60 * 60 = 86400 samples. The json is not being streamed, but formed as a single object in memory, but I should not think that is an issue with this amount of data.

**Comment by sbender9** [comment]
I'm using curl to test. 

`
$ curl "http://localhost:3000/signalk/v1/api/self/track?timespan=1d&resolution=1s"
{"type":"MultiLineString","coordinates":[]}$ 
`

`curl "http://localhost:3000/signalk/v1/api/self/track?timespan=1d&resolution=1m"
{"type":"MultiLineString","coordinates":[[[-76.4627452,39.0695028],[-76.4622588,39.0692838]],[[-76.4640445,39.070093],[-76.4640445,39.070093],[-76.4631901,39.0697008],[-76.4617232,39.0690472]],[[-76.4635482,39.069863]],[[-76.4342495,39.0566583],[-76.4330107,39.056094],[-76.4314491,39.0555408],[-76.4297438,39.0553073],[-76.4280285,39.0551305],[-76.4263111,39.0549589],[-76.4246096,39.0547953],[-76.4229064,39.0546305],[-76.4215763,39.0545589],[-76.4211915,39.0535516],[-76.4208081,39.052534],[-76.4201515,39.0516628],[-76.4188142,39.0522602],[-76.4173091,39.0526507],[-76.4157933,39.0526819],[-76.4148352,39.0527214],[-76.4140154,39.0527693],[-76.4131075,39.0528107],[-76.4121981,39.0528567],[-76.411077,39.0529234],[-76.409856,39.052984],[-76.4085662,39.0530557],[-76.4073703,39.0531166],[-76.4062033,39.0531755]],[[-76.4634428,39.0698152],[-76.4620036,39.0691745],[-76.4605629,39.0685192],[-76.4591396,39.0678684],[-76.4576981,39.0672292],[-76.4562455,39.0665757],[-76.4548393,39.06594],[-76.4534604,39.0653172],[-76.4520931,39.0647025],[-76.4507423,39.064091],[-76.4494146,39.0634917],[-76.4480939,39.0628984],[-76.4467368,39.0622847],[-76.4453601,39.0616606],[-76.4439365,39.0610164],[-76.4424902,39.0603719]]]}
`

**Comment by sbender9** [comment]
Why not just a check box for "Record Track" in config ui?

**Comment by tkurki** [comment]
I'm really swamped, so merging & publishing - PR the track recording switch, please, a good idea.

**Pull request merged**